### PR TITLE
syntax: rework syntax structure

### PR DESCRIPTION
**Pull request by l-zeuch**
This commit reworks all syntax files to be less cluttered and more streamlined.
Moreover, the main file is moved one directory up and renamed to yagpdbcc.vim.
This way, we have a better folder structure to follow along.

Test cases have been updated to reflect the changes of the syntax groups, as
their names are now considerably shorter.

Moreover, the gen_code_completion.py file has been updated as well to work with
this new folder structure, and the completion sources have been updated along.

All tests are passing locally.

**Terms**
- [x] I agree to follow this project's [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I have read and understood this project's [Contributing Guidelines](../CONTRIBUTING.md)

### CODE REVIEW

File: lua/yagpdbcc.lua [hunk 1]
```diff
@@ -45,35 +45,6 @@ end
...
-		{ label = 'sub' },
-		{ label = 'mult' },
-		{ label = 'div' },
-		{ label = 'fdiv' },
-		{ label = 'log' },
-		{ label = 'mod' },
-		{ label = 'pow' },
-		{ label = 'sqrt' },
-		{ label = 'nil' },
-		{ label = 'end' },
```
Comment (LRitzdorf) [inline lua/yagpdbcc.lua:76]: Did we mean to remove these? If not, the changes to `gen_code_completion.py` might not be working quite as expected.

Comment (l-zeuch) [reply lua/yagpdbcc.lua:76]: It's actually the changes made to `syntax/yagpdbcc.vim` which are causing this... I'll move them where they belong.

**Comment by l-zeuch** [comment]
There are still some functions missing, I must've forgotten to actually add them. New commit follows.

**Comment by l-zeuch** [comment]
In `lua/yagpdbcc.lua` were some duplicates before, which is why the diff there shows more deletions than additions. The additions are given by the fact that these declarations are now included at a later point of traversing the syntax files.

File: syntax/yagpdbcc/functions.vim [hunk 2]
```diff
@@ -16,90 +16,110 @@
 " with this program; if not, write to the Free Software Foundation, Inc.,
 " 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-" Language: YAGPDB Custom Commands
-" Maintainer: Luca Zeuch <l-zeuch@email.de>,
-"   LRitzdorf <42657792+LRitzdorf@users.noreply.github.com>
+" Language: YAGPDB Custom Commands Maintainer: Luca Zeuch <l-zeuch@email.de>, LRitzdorf <42657792+LRitzdorf@users.noreply.github.com>
```
Comment (LRitzdorf) [inline syntax/yagpdbcc/functions.vim:19]: Minor line wrapping issue (not a big deal)

**Review by LRitzdorf — APPROVED** [review]
Looks good! This did really need some cleanup — thanks for taking care of that!

**Pull request merged**